### PR TITLE
Vijay master module

### PR DESCRIPTION
**Pull request by ajax-vijayherage**
tehsil and city service changes done

### CODE REVIEW

File: src/main/java/com/ajax/controller/CityController.java [hunk 1]
```diff
@@ -32,24 +35,27 @@
...
-                                                    throws Exception {
-        service.deleteCityById(id);
-        return HttpStatus.FORBIDDEN;
-    }
+
+    @DeleteMapping("/delete/{code}")
+	public HttpStatus deleteCityByCode(@PathVariable("code") String cityCode) throws Exception {
+    	CityEntity cityEntity = service.getCityByCode(cityCode);
+    	cityEntity.setIsActive(CommonConstants.INACTIVE_STS);
+		service.deleteCityByCode(cityEntity);
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/controller/CityController.java]: Code format

File: src/main/java/com/ajax/controller/ItemCategouryController.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import com.ajax.model.ItemCategoury;
+import com.ajax.service.ItemCategouryService;
+import com.ajax.transform.ItemCategouryTransformer;
+import com.ajax.util.CommonConstants;
+
+@RestController
+@RequestMapping("/ItemCategoury")
+public class ItemCategouryController {
+
+	@Autowired
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/controller/ItemCategouryController.java]: Code format

File: src/main/java/com/ajax/controller/StateController.java [hunk 3]
```diff
@@ -1,58 +1,65 @@
...
+  @RequestMapping("/State") 
+  public class StateController {
+  
+  
+  @Autowired StateService service;
+  
+  @GetMapping("/getList")
+  public ResponseEntity<List<State>> getAllState()
+  {
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/controller/StateController.java]: Remove extra line

File: src/main/java/com/ajax/controller/StateController.java [hunk 4]
```diff
@@ -1,58 +1,65 @@
...
+  public ResponseEntity<State> createOrUpdateState(@RequestBody State state) throws Exception
+  { 
+	  State updated = StateTransformer.getState(service.createOrUpdateState(state)); 
+	  return new ResponseEntity<State>(updated, new HttpHeaders(), HttpStatus.OK); 
+  }
+  
+  @DeleteMapping("/delete/{code}") 
+  public HttpStatus deleteStateByCode(@PathVariable("code") String stateCode) throws Exception 
+  {
+		  StateEntity stateEntity = service.getStateByCode(stateCode);
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/controller/StateController.java]: Code format

File: src/main/java/com/ajax/controller/TaxCategoryController.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import com.ajax.model.TaxCategory;
+import com.ajax.service.TaxCategoryService;
+import com.ajax.transform.TaxCategoryTransformer;
+import com.ajax.util.CommonConstants;
+
+@RestController
+@RequestMapping("/TaxCategory")
+public class TaxCategoryController {
+
+	@Autowired
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/controller/TaxCategoryController.java]: Code format

File: src/main/java/com/ajax/controller/TaxCategoryController.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	TaxCategoryService service;
+ 
+    @GetMapping("/getList")
+    public ResponseEntity<List<TaxCategory>> getAllTaxCategory() {
+        List<TaxCategory> list = TaxCategoryTransformer.getTaxCategoryList(service.getAllTaxCategory());
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/controller/TaxCategoryController.java]: Remove extra line

File: src/main/java/com/ajax/controller/TaxRateController.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import com.ajax.model.TaxRate;
+import com.ajax.service.TaxRateService;
+import com.ajax.transform.TaxRateTransformer;
+import com.ajax.util.CommonConstants;
+
+@RestController
+@RequestMapping("/TaxRate")
+public class TaxRateController {
+
+	@Autowired
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/controller/TaxRateController.java]: Code format

File: src/main/java/com/ajax/controller/TehsilController.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+@RequestMapping("/Tehsil")
+public class TehsilController {
+
+	@Autowired
+	TehsilService service;
+ 
+    @GetMapping("/getList")
+    public ResponseEntity<List<Tehsil>> getAllTehsil() {
+        List<Tehsil> list = TehsilTransformer.getTehsilList(service.getAllTehsil());
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/controller/TehsilController.java]: Remove extra line

File: src/main/java/com/ajax/serviceImpl/TehsilServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+			tehsilEntity.setIsActive(entity.getIsActive());
+			Optional<TehsilEntity> optTehsil = Optional.of(tehsilEntity);
+			if (optTehsil.isPresent()) {
+				repository.save(tehsilEntity);
+				return tehsilEntity;
+			}
+			return null;
+		}
+	}
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/serviceImpl/TehsilServiceImpl.java]: Remove extra line

File: src/main/java/com/ajax/serviceImpl/CityServiceImpl.java [hunk 10]
```diff
@@ -29,45 +30,53 @@
...
+		CityEntity cityEntity = repository.findByCityCode(cityCode);
+		Optional<CityEntity> optCity = Optional.of(cityEntity);
 
+		if (optCity.isPresent()) {
+			return optCity.get();
+		} else {
+			throw new Exception("No City record exist for given id");
+		}
+	}
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/serviceImpl/CityServiceImpl.java]: Remove extra line

File: src/main/java/com/ajax/serviceImpl/BrandServiceImpl.java [hunk 11]
```diff
@@ -28,45 +31,51 @@
...
-        if(brand.isPresent()) 
-        {
-            repository.deleteById(id);
-        } else {
-            throw new Exception("No Brand record exist for given id");
-        }
-    } 
+        brandEntity.setModifiedDate(CommonUtils.getCurrentDateTime());
+		repository.save(brandEntity);
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/serviceImpl/BrandServiceImpl.java]: Remove extra line

File: src/main/java/com/ajax/service/TaxRateService.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+import com.ajax.enity.TaxRateEntity;
+import com.ajax.model.TaxRate;
+
+public interface TaxRateService {
+
+	  public List<TaxRateEntity> getAllTaxRate();
+     
+	    public TaxRateEntity getTaxRateByCode(String taxRateCode) throws Exception;
+	     
+	    public TaxRateEntity createOrUpdateTaxRate(TaxRate taxRate) throws Exception;
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/service/TaxRateService.java]: Code format

File: src/main/java/com/ajax/service/TaxCategoryService.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
+package com.ajax.service;
+
+import java.util.List;
+
+import com.ajax.enity.TaxCategoryEntity;
+import com.ajax.model.TaxCategory;
+
+public interface TaxCategoryService {
+
+	public List<TaxCategoryEntity> getAllTaxCategory();
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/service/TaxCategoryService.java]: Code format

File: src/main/java/com/ajax/service/ItemGroupService.java [hunk 14]
```diff
@@ -0,0 +1,17 @@
+package com.ajax.service;
+
+import java.util.List;
+
+import com.ajax.enity.ItemGroupEntity;
+import com.ajax.model.ItemGroup;
+
+public interface ItemGroupService {
+
+	public List<ItemGroupEntity> getAllItemGroup();
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/service/ItemGroupService.java]: Code format

File: src/main/java/com/ajax/service/ItemCategouryService.java [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+
+public interface ItemCategouryService {
+
+	 public List<ItemCategouryEntity> getAllItemCategoury();
+     
+	    public ItemCategouryEntity getItemCategouryByCode(String itemCategouryCode) throws Exception;
+	     
+	    public ItemCategouryEntity createOrUpdateItemCategoury(ItemCategoury itemCategoury) throws Exception; 
+	     
+		public void deleteItemCategouryByCode(ItemCategouryEntity itemCategouryEntity) throws Exception;
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/service/ItemCategouryService.java]: Code format

File: src/main/java/com/ajax/model/Tehsil.java [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	
+	
+	@Override
+	public String toString() {
+		return "Tehsil [tehsilCode=" + tehsilCode + ", tehsilNameEng=" + tehsilNameEng
+				+ ", tehsilNameUni=" + tehsilNameUni + ", districtCode=" + districtCode + ", isActive=" + isActive+ "]";
+	}
+
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/model/Tehsil.java]: Remove extra line

File: src/main/java/com/ajax/model/TaxRate.java [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+		this.isActive = isActive;
+		return this;
+	}
+
+	@Override
+	public String toString() {
+		return "TaxRate [taxRateCode=" + taxRateCode + ", taxCategoryCode=" + taxCategoryCode + ", taxRateDescEng="
+				+ taxRateDescEng + ", stateCode=" + stateCode + ", isActive=" + isActive + "]";
+	}
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/model/TaxRate.java]: Remove extra line

File: src/main/java/com/ajax/enity/TehsilEntity.java [hunk 18]
```diff
@@ -0,0 +1,185 @@
...
+		this.createdBy = createdBy;
+		this.createdDate = createdDate;
+		this.modifiedBy = modifiedBy;
+		this.modifiedDate = modifiedDate;
+	}
+	
+	public TehsilEntity() {
+		// TODO Auto-generated constructor stub
+	}
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/enity/TehsilEntity.java]: Remove extra line

File: src/main/java/com/ajax/enity/TaxCategoryEntity.java [hunk 19]
```diff
@@ -0,0 +1,151 @@
...
+
+	@Override
+	public String toString() {
+		return "TaxCategoryEntity [ taxCategoryCode=" + taxCategoryCode + ", taxCategoryDesc="
+				+ taxCategoryDesc + ", sortBy=" + sortBy + ", isActive=" + isActive + ", createdBy=" + createdBy
+				+ ", createdDate=" + createdDate + ", modifiedBy=" + modifiedBy + ", modifiedDate=" + modifiedDate
+				+ "]";
+	}
+
+
```
Comment (ajax-rahul) [inline src/main/java/com/ajax/enity/TaxCategoryEntity.java]: Remove extra line

Comment (ajax-vijayherage) [reply src/main/java/com/ajax/enity/TaxCategoryEntity.java]: Resolve

File: src/main/java/com/ajax/enity/TehsilEntity.java (hunk 18 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/enity/TehsilEntity.java]: Resolve

File: src/main/java/com/ajax/model/TaxRate.java (hunk 17 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/model/TaxRate.java]: Resolve

File: src/main/java/com/ajax/model/Tehsil.java (hunk 16 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/model/Tehsil.java]: Resolve

File: src/main/java/com/ajax/service/ItemCategouryService.java (hunk 15 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/service/ItemCategouryService.java]: Resolve

File: src/main/java/com/ajax/controller/TaxCategoryController.java (hunk 5 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/controller/TaxCategoryController.java]: Resolve

File: src/main/java/com/ajax/controller/StateController.java (hunk 4 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/controller/StateController.java]: Resolve

File: src/main/java/com/ajax/controller/StateController.java (hunk 3 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/controller/StateController.java]: Resolve

File: src/main/java/com/ajax/controller/ItemCategouryController.java (hunk 2 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/controller/ItemCategouryController.java]: Resolve

File: src/main/java/com/ajax/controller/CityController.java (hunk 1 above)
Comment (ajax-vijayherage) [reply src/main/java/com/ajax/controller/CityController.java]: Resolve

**Pull request merged**